src/status_im/contexts/wallet/add_account/add_address_to_watch/view.cljs

```diff
@@ -2,168 +2,82 @@
-
-(defn activity-indicator
-  [activity-state]
-  (let [{:keys [message]
-         :as   props} (case activity-state
-                        :has-activity               {:accessibility-label :account-has-activity
-                                                     :icon                :i/done
-                                                     :type                :success
-                                                     :message             :t/address-activity}
-                        :no-activity                {:accessibility-label :account-has-no-activity
-                                                     :icon :i/info
-                                                     :type :warning
-                                                     :message :t/this-address-has-no-activity}
-                        :invalid-ens                {:accessibility-label :error-message
-                                                     :icon                :i/info
-                                                     :type                :error
-                                                     :message             :t/invalid-address}
-                        :address-already-registered {:accessibility-label :error-message
-                                                     :icon                :i/info
-                                                     :type                :error
-                                                     :message             :t/address-already-in-use}
-                        {:accessibility-label :searching-for-activity
-                         :icon                :i/pending-state
-                         :type                :default
-                         :message             :t/searching-for-activity})]
-    (when activity-state
-      [quo/info-message
-       (assoc props
-              :style style/info-message
-              :size  :default)
-       (i18n/label message)])))
+(defn- on-press-confirm-add-address-to-watch
+  [input-value]
+  (rf/dispatch
+   [:wallet/confirm-add-address-to-watch
+    {:address input-value
+     :ens?    (utils.ens/is-valid-eth-name?
+               input-value)}]))
 
 (defn view
   []
-  (let [addresses           (rf/sub [:wallet/addresses])
-        input-value         (reagent/atom nil)
-        validate            #(validate-address (set addresses) %)
-        validation-msg      (reagent/atom nil)
-        clear-input         (fn []
-                              (reset! input-value nil)
-                              (reset! validation-msg nil)
-                              (rf/dispatch [:wallet/clear-address-activity])
-                              (rf/dispatch [:wallet/clean-scanned-address]))
+  (let [addresses (rf/sub [:wallet/lowercased-addresses])
+        {:keys [title description input-title adding-address-purpose accessibility-label]}
+        (rf/sub [:wallet/currently-added-address])
+        validate #(wallet.utils/validate-fn % addresses)
```

Comment:
Exactly same logic as the previous `add_address_to_save` file, But I had to break it up into two files to make a separation of concerns between them in case we ever had to update one or the other.